src/components/timeout-dialog/timeout-dialog.js

```diff
@@ -267,8 +283,8 @@ function TimeoutDialog($module) {
   const keepAliveAndClose = () => {
     cleanup();
     setupDialogTimer();
-    utils.ajaxGet(settings.keepAliveUrl, () => {
-    });
+    broadcastSessionActivity();
+    utils.ajaxGet(settings.keepAliveUrl, () => {});
```

Comment:
@kyle-bowden / @ellamdav should broadcastSessionActivity following the keep alive request? The timeout dialog doesn't actually handle the keep alive request failing, so might not make a difference in practice

Reply:
yeah, we considered that, but decided to decouple the two because there's no existing error handling, and they seemed like independent fire-and-forget events.  also, it would be more complicated to test the callback from `ajaxGet`.  do you think we need to consider what should happen if the ajax keepalive call were to fail?

Reply:
I think the timeout dialog should probably handle it, but not something worth adding in this change, just a retry would go some distance - or showing an error - at the moment it would act like it kept the session alive but if it failed (in certain ways) then it wouldn't have actually extended the session and the next time user tried to do something they might get an error. It's the kind of thing that I think would be much better defined as an explicit state machine https://xstate.js.org/docs/ because I bet there's a load of invalid states it can be in at the moment that haven't been thought about